x/attribute/keeper/keeper.go

```diff
@@ -245,22 +245,23 @@ func (k Keeper) UpdateAttribute(ctx sdk.Context, originalAttribute types.Attribu
 		return fmt.Errorf("%q does not resolve to address %q", updateAttribute.Name, owner.String())
 	}
 
+	store := ctx.KVStore(k.storeKey)
 	addrBz := originalAttribute.GetAddressBytes()
+	attrKey := types.AddrAttributeKey(addrBz, originalAttribute)
+	currentAttr := store.Get(attrKey)
 
-	store := ctx.KVStore(k.storeKey)
-	it := sdk.KVStorePrefixIterator(store, types.AddrAttributesNameKeyPrefix(addrBz, normalizedOrigName))
-	defer it.Close()
 	var found bool
-	for ; it.Valid(); it.Next() {
+	if currentAttr != nil {
```

Comment:
Probably safer to use `len(currentAttr) > 0` here.

It's best to use that check for usability of a slice unless it's a situation where you want to treat a nil slice differently from an empty one. You can't always count on functions to return nil instead of an empty slice (or vice versa), and usually we only care if it returned something. So it's best practice to `len(thing) > 0` instead of `thing != nil`

Reply:
I was thinking that the `nil` return here is actually a not found.  
```
// Get returns nil if key doesn't exist. Panics on nil key.
	Get(key []byte) []byte
```

A return of an empty slice would mean the key did exist with an empty slice.  This should hit the error here: 
```
attr := types.Attribute{}
		if err := k.cdc.Unmarshal(currentAttr, &attr); err != nil {
			return err
		}
```

This would indicate something bad happened else where in the code to have set this to an empty slice.  The error here would be helpful.  `len(currentAttr) > 0` would just hide it.

Reply:
I just did a quick look through the codebase for the attribute module, and I don't **think** I found a case where the currentAttr could be 0. It may make the code a bit easier to understand if we use **store.Has** or a **nil check** because checking the length will imply that we store empty attributes somewhere.